[Bug]: gnome-font-viewer build failure

### Problem description


[109/112] Compiling C object src/gnome-font-viewer.p/font-view.c.o
FAILED: src/gnome-font-viewer.p/font-view.c.o 
aarch64-linux-android-clang -Isrc/gnome-font-viewer.p -Isrc -I../source/src -I. -I../source -I/data/data/com.termux/files/usr/include/glib-2.0 -I/data/data/com.termux/files/usr/lib/glib-2.0/include -I/data/data/com.termux/files/usr/include -I/data/data/com.termux/files/usr/include/gtk-3.0 -I/data/data/com.termux/files/usr/include/gio-unix-2.0 -I/data/data/com.termux/files/usr/include/cairo -I/data/data/com.termux/files/usr/include/pango-1.0 -I/data/data/com.termux/files/usr/include/fribidi -I/data/data/com.termux/files/usr/include/harfbuzz -I/data/data/com.termux/files/usr/include/atk-1.0 -I/data/data/com.termux/files/usr/include/pixman-1 -I/data/data/com.termux/files/usr/include/libxml2 -I/data/data/com.termux/files/usr/include/freetype2 -I/data/data/com.termux/files/usr/include/gdk-pixbuf-2.0 -I/data/data/com.termux/files/usr/include/libpng16 -I/data/data/com.termux/files/usr/include/libhandy-1 -I/data/data/com.termux/files/usr/include/gsettings-desktop-schemas -I/data/data/com.termux/files/usr/include/gnome-desktop-3.0 -fcolor-diagnostics -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Oz -g -fstack-protector-strong -Oz -fPIE -pthread -MD -MQ src/gnome-font-viewer.p/font-view.c.o -MF src/gnome-font-viewer.p/font-view.c.o.d -o src/gnome-font-viewer.p/font-view.c.o -c ../source/src/font-view.c
../source/src/font-view.c:210:65: error: too few arguments to function call, expected 5, have 3
        (factory, thumb_uri, g_file_info_get_content_type (info));
                                                                ^
/data/data/com.termux/files/usr/include/gnome-desktop-3.0/libgnome-desktop/gnome-desktop-thumbnail.h:84:13: note: 'gnome_desktop_thumbnail_factory_generate_thumbnail' declared here
GdkPixbuf * gnome_desktop_thumbnail_factory_generate_thumbnail (GnomeDesktopThumbnailFactory *factory,
            ^
../source/src/font-view.c:219:82: error: too few arguments to function call, expected 6, have 4
                                                        thumb_uri, (time_t) mtime);
                                                                                 ^
/data/data/com.termux/files/usr/include/gnome-desktop-3.0/libgnome-desktop/gnome-desktop-thumbnail.h:89:24: note: 'gnome_desktop_thumbnail_factory_save_thumbnail' declared here
gboolean               gnome_desktop_thumbnail_factory_save_thumbnail (GnomeDesktopThumbnailFactory *factory,
                       ^
../source/src/font-view.c:224:91: error: too few arguments to function call, expected 5, have 3
                                                                 thumb_uri, (time_t) mtime);
                                                                                          ^
/data/data/com.termux/files/usr/include/gnome-desktop-3.0/libgnome-desktop/gnome-desktop-thumbnail.h:95:24: note: 'gnome_desktop_thumbnail_factory_create_failed_thumbnail' declared here
gboolean               gnome_desktop_thumbnail_factory_create_failed_thumbnail (GnomeDesktopThumbnailFactory *factory,
                       ^
3 errors generated.
[110/112] Compiling C object src/gnome-font-viewer.p/sushi-font-widget.c.o
../source/src/sushi-font-widget.c:98:14: warning: 'pango_find_base_dir' is deprecated [-Wdeprecated-declarations]
  base_dir = pango_find_base_dir (text, -1);
             ^
/data/data/com.termux/files/usr/include/pango-1.0/pango/pango-bidi-type.h:106:1: note: 'pango_find_base_dir' has been explicitly marked deprecated here
PANGO_DEPRECATED_IN_1_44
^
/data/data/com.termux/files/usr/include/pango-1.0/pango/pango-version-macros.h:687:49: note: expanded from macro 'PANGO_DEPRECATED_IN_1_44'
# define PANGO_DEPRECATED_IN_1_44               PANGO_DEPRECATED
                                                ^
/data/data/com.termux/files/usr/include/pango-1.0/pango/pango-version-macros.h:385:49: note: expanded from macro 'PANGO_DEPRECATED'
# define PANGO_DEPRECATED                       G_DEPRECATED _PANGO_EXTERN
                                                ^
/data/data/com.termux/files/usr/include/glib-2.0/glib/gmacros.h:1201:37: note: expanded from macro 'G_DEPRECATED'
#define G_DEPRECATED __attribute__((__deprecated__))
                                    ^
1 warning generated.
[111/112] Generating data/org.gnome.font-viewer.appdata.xml with a custom command
ninja: build stopped: subcommand failed.


### What steps will reproduce the bug?

build

### What is the expected behavior?

_No response_

### System information

termux-info:

```

```

--- a/x11-packages/gnome-font-viewer/build.sh
+++ b/x11-packages/gnome-font-viewer/build.sh
@@ -5,6 +5,7 @@ TERMUX_PKG_MAINTAINER="@termux"
 # Stick to GTK 3 for now.
 _MAJOR_VERSION=41
 TERMUX_PKG_VERSION=${_MAJOR_VERSION}.0
+TERMUX_PKG_REVISION=1
 TERMUX_PKG_SRCURL=https://download.gnome.org/sources/gnome-font-viewer/${_MAJOR_VERSION}/gnome-font-viewer-${TERMUX_PKG_VERSION}.tar.xz
 TERMUX_PKG_SHA256=5dd410331be070e4e034397f2754980e073851d50a2119f2fbf96adc6fe2e876
 TERMUX_PKG_DEPENDS="fontconfig, freetype, gdk-pixbuf, glib, gnome-desktop3, gtk3, harfbuzz, libcairo, libhandy, pango"

--- a/x11-packages/gnome-font-viewer/fix-build-against-gnome-desktop-42.0.patch
+++ b/x11-packages/gnome-font-viewer/fix-build-against-gnome-desktop-42.0.patch
@@ -1,0 +1,28 @@
+--- a/src/font-view.c
++++ b/src/font-view.c
+@@ -207,7 +207,7 @@
+ 
+     factory = gnome_desktop_thumbnail_factory_new (GNOME_DESKTOP_THUMBNAIL_SIZE_LARGE);
+     pixbuf = gnome_desktop_thumbnail_factory_generate_thumbnail
+-        (factory, thumb_uri, g_file_info_get_content_type (info));
++        (factory, thumb_uri, g_file_info_get_content_type (info), NULL, NULL);
+ 
+     mtime = g_file_info_get_attribute_uint64 (info, G_FILE_ATTRIBUTE_TIME_MODIFIED);
+     if (pixbuf != NULL) {
+@@ -216,12 +216,14 @@
+                                                      128 * scale_factor,
+                                                      GDK_INTERP_BILINEAR);
+         gnome_desktop_thumbnail_factory_save_thumbnail (factory, pixbuf,
+-                                                        thumb_uri, (time_t) mtime);
++                                                        thumb_uri, (time_t) mtime,
++                                                        NULL, NULL);
+         g_object_unref (pixbuf);
+         pixbuf = scaled;
+     } else {
+         gnome_desktop_thumbnail_factory_create_failed_thumbnail (factory,
+-                                                                 thumb_uri, (time_t) mtime);
++                                                                 thumb_uri, (time_t) mtime,
++                                                                 NULL, NULL);
+     }
+ 
+     return g_steal_pointer (&pixbuf);

